### PR TITLE
JEN-795 Failed connect to vault.public-ci.percona.com:8600

### PR DESCRIPTION
**Pull request by robgolebiowski**
vault.public-ci.percona was deemed to be unsecure and thus removed.
keyring_vault MTR test should use https://vault.percona.com, which is
maintained by IIT team. The tests were amended to use this new server.
The policy was changed to more strict one in comparision to
vault.public-ci.percona - the tests were also amended to cope with the
new policy.

Also new tests were added to test the separation of keyrings inside a
Vault server on directory level. For those tests only one new mount
point is created on the server and the separation is done by creating
separate directories inside the mount point - one directory per each
Percona server keyring_vault keyring.

The old tests are still checking the separation of keyrings with use of
separate mount points.

The not_repeat.inc was added to check whether the tests are not run with
--repeat > 1. This is because the tests are creating mount points at
the beginning of a run and delete when at the end of the run. When repeat is
used server will remember that it has already created master key and will
try to fetch the key. Unfortunately the key will be gone because the
mount point was deleted and re-created.

Changed the number of times the Master Key is rotated in
keyring_vault.table_encrypt_3.test and
keyring_vault.table_encrypt_3_directory to 100, so not to DoS the Vault Server.

Ported keyring_vault unittest to 8.0 - as they were not building,
enabled MERGE_UNITTESTS for keyring_vault as the linking of unittests
in 8.0 takes substantial amount of time.

Also clang-format-5.0 was applied.

(cherry picked from commit 78e4297ff14fa0f419fb29e8168399a24e9fca7c)

### CODE REVIEW

**Comment by robgolebiowski** [comment]
https://ps2.cd.percona.com/view/8.0/job/percona-server-8.0-param/56/

File: plugin/keyring_vault/vault_curl.cc [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 namespace keyring {
 
-static const constexpr size_t max_response_size = 32000000;
```
Comment (laurynas-biveinis) [inline plugin/keyring_vault/vault_curl.cc:31]: Why?

File: plugin/keyring_vault/vault_curl.cc [hunk 2]
```diff
@@ -167,7 +167,7 @@ bool Vault_curl::setup_curl_session(CURL *curl) {
 
   last_ping_time = my_timer_milliseconds();
 
-  if ((list = curl_slist_append(list, token_header.c_str())) == nullptr ||
```
Comment (laurynas-biveinis) [inline plugin/keyring_vault/vault_curl.cc]: Why?

File: unittest/gunit/keyring_vault/CMakeLists.txt [hunk 3]
```diff
@@ -50,40 +39,87 @@ SET(TESTS
...
+  ENDIF()
+  ADD_DEPENDENCIES(merge_keyring_vault_tests-t GenError)
+
+ELSE(MERGE_UNITTESTS)
+
+  FOREACH(test ${TESTS})
+    MYSQL_ADD_EXECUTABLE(${test}-t ${test}-t.cc SKIP_INSTALL)
+    TARGET_LINK_LIBRARIES(${test}-t keyring_vault_test)
+
+    #TARGET_LINK_LIBRARIES(${test}-t gunit_large strings dbug regex)
```
Comment (laurynas-biveinis) [inline unittest/gunit/keyring_vault/CMakeLists.txt]: Remove commented-out block

File: unittest/gunit/keyring_vault/vault_io-t.cc [hunk 4]
```diff
@@ -314,14 +315,18 @@ TEST_F(Vault_io_test, FlushKeyRetrieveDeleteInit) {
...
 class Mock_vault_curl : public IVault_curl {
  public:
   MOCK_METHOD1(init, bool(const Vault_credentials &vault_credentials));
   MOCK_METHOD1(list_keys, bool(Secure_string *response));
   MOCK_METHOD2(write_key, bool(const Vault_key &key, Secure_string *response));
   MOCK_METHOD2(read_key, bool(const Vault_key &key, Secure_string *response));
   MOCK_METHOD2(delete_key, bool(const Vault_key &key, Secure_string *response));
-  MOCK_METHOD1(set_timeout, void(uint timeout));
+  MOCK_NOEXCEPT_METHOD1(set_timeout, void(uint timeout));
+  // MOCK_METHOD1(set_timeout, void(uint timeout));
```
Comment (laurynas-biveinis) [inline unittest/gunit/keyring_vault/vault_io-t.cc]:
```suggestion
```

File: unittest/gunit/keyring_vault/vault_keyring-api-t.cc [hunk 5]
```diff
@@ -312,84 +318,84 @@ TEST_F(Keyring_vault_api_test, NullUser) {
...
   EXPECT_TRUE(mysql_key_generate("", "AES", "Robert", 128));
   EXPECT_CALL(*(reinterpret_cast<Mock_logger *>(logger.get())),
-              log(MY_ERROR_LEVEL,
+              log(MY_WARNING_LEVEL,
                   StrEq("Error while generating key: key_id cannot be empty")));
   EXPECT_TRUE(mysql_key_generate("", "AES", nullptr, 128));
 }
 
-TEST_F(Keyring_vault_api_test, StorePBStoreSKFetchPBRemovePB) {
+TEST_F(Keyring_vault_api_test, StorePBStoreSKFetchPBRemovePerconInnodb) {
```
Comment (laurynas-biveinis) [inline unittest/gunit/keyring_vault/vault_keyring-api-t.cc]:
```suggestion
TEST_F(Keyring_vault_api_test, StorePBStoreSKFetchPBRemovePerconaInnodb) {
```

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
Minor comments only, looks great overall

File: plugin/keyring_vault/vault_curl.cc (hunk 2 above)
Comment (robgolebiowski) [reply plugin/keyring_vault/vault_curl.cc]: A mistake - probably during merging while cherry-picking, reverted.

File: plugin/keyring_vault/vault_curl.cc (hunk 1 above)
Comment (robgolebiowski) [reply plugin/keyring_vault/vault_curl.cc:31]: Because constexpr implies const.

**Review by laurynas-biveinis — APPROVED** [review]

File: plugin/keyring_vault/tests/mtr/innodb_row_log_encryption.test [hunk 6]
```diff
@@ -10,17 +11,18 @@
...
-call mtr.add_suppression("\\[ERROR\\] Function 'keyring_vault' already exists");
-call mtr.add_suppression("\\[ERROR\\] Couldn't load plugin named 'keyring_vault' with soname 'keyring_vault.*'.");
+call mtr.add_suppression("Function 'keyring_vault' already exists");
+call mtr.add_suppression("Couldn't load plugin named 'keyring_vault' with soname 'keyring_vault.*'.");
 call mtr.add_suppression("Plugin keyring_vault reported");
+call mtr.add_suppression("keyring_vault initialization failure");
 
---replace_result $MYSQLTEST_VARDIR MYSQLTEST_VARDIR
-eval SET @@global.keyring_vault_config="$KEYRING_CONF_FILE_1";
+#--replace_result $MYSQLTEST_VARDIR MYSQLTEST_VARDIR
```
Comment (percona-ysorokin) [inline plugin/keyring_vault/tests/mtr/innodb_row_log_encryption.test:19]: Commented out code.

File: unittest/gunit/keyring_vault/uuid.cc [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  boost::uuids::uuid uuid = gen();
+  std::ostringstream uuid_ss;
+  uuid_ss << uuid;
+  return uuid_ss.str();
+}
+
+std::string get_key_signature(const std::string &uuid,
+                              const std::string &key_id,
+                              const std::string &user) {
+  std::string id = uuid + key_id;
```
Comment (percona-ysorokin) [inline unittest/gunit/keyring_vault/uuid.cc:18]:
```suggestion
```

File: unittest/gunit/keyring_vault/uuid.cc [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  std::ostringstream signature;
+  signature << id.length() << '_' << id << user.length() << '_' << user;
```
Comment (percona-ysorokin) [inline unittest/gunit/keyring_vault/uuid.cc:20]: Just a suggestion to simplify this a bit
```suggestion
  signature << uuid.length() + key_id.length() << '_' << uuid << key_id << user.length() << '_' << user;
```

**Review by percona-ysorokin — COMMENTED** [review]
The same comments/questions as in 5.7 + commented out code

**Comment by percona-ysorokin** [comment]
Another question is what's the reason of introducing `MERGE_UNITTESTS` define?

**Comment by laurynas-biveinis** [comment]
As discussed on Slack, merging now for better Vault testing, Yura's review comments will be addressed once Robert returns

**Pull request merged**